### PR TITLE
Provide safer temporary files and directories

### PR DESCRIPTION
**Pull request by lbonn**
As mentioned in PRO-4721 and PRO-4722, TemporaryFile and
TemporaryDirectory were somewhat unsafe.

To solve this, we create a global temporary directory root for the
lifetime of the application, with restrictive permissions. The directory
is created in the OS' canonical temporary directory so that aktualizr
can eventually run under a lesser privileged user.

### CODE REVIEW

**Comment by lbonn** [comment]
I added some bloat to remove the directory at the end of the program. Not critical, but nice anyway, especially in dev environments.

File: src/utils.cc [hunk 1]
```diff
@@ -330,8 +334,47 @@ boost::filesystem::path Utils::absolutePath(const boost::filesystem::path &root,
...
+  }
+  ~SafeTempRoot() {
+    try {
+      boost::filesystem::remove_all(root);
+    } catch (...) {
+      // ignore this, not critical
+    }
+  }
+
+  boost::filesystem::path root;
```
Comment (patriotyk) [inline src/utils.cc]: maybe 'path' is better name?

Comment (lbonn) [reply src/utils.cc]: Yes you're right, I made the change.

File: src/utils.cc [hunk 2]
```diff
@@ -330,8 +334,47 @@ boost::filesystem::path Utils::absolutePath(const boost::filesystem::path &root,
   }
 }
 
+class SafeTempRoot : private boost::noncopyable {
+ public:
+  // provide this as a static method so that we can use C++ static destructor
+  // to remove the temp root
+  static boost::filesystem::path &Get() {
+    {
+      // Ensure thread safety (uneeded in c++11)
```
Comment (patrickvacek) [inline src/utils.cc]: +1 for leaving a note so that we can clean up once we upgrade for good.

**Comment by patrickvacek** [comment]
This is cool and looks good to me. I recommend that we have @heartsucker approve this before merging, though.

**Pull request merged**